core/web/middleware.go

```diff
@@ -0,0 +1,194 @@
+	return file, info, nil
+}
+
+// List of encodings we would prefer to use, in order of preference, best first.
+// We only support gzip for now
+var preferredEncodings = []string{"gzip"}
+
+// File extension to use for different encodings.
+func extensionForEncoding(encname string) string {
+	switch encname {
+	case "gzip":
+		return ".gz"
+	}
+	return ""
+}
+
+// Find the best file to serve based on the client's Accept-Encoding, and which
+// files actually exist on the filesystem. If no file was found that can satisfy
+// the request, the error field will be non-nil.
+func (f *gzipFileHandler) findBestFile(w http.ResponseWriter, r *http.Request, fpath string) (http.File, os.FileInfo, error) {
+	ae := r.Header.Get(acceptEncodingHeader)
+	// Send the base file if no AcceptEncoding header is provided
+	if ae == "" {
+		return f.openAndStat(fpath)
+	}
+
+	// Got an accept header? See what possible encodings we can send by looking for files
+	var available []string
+	for _, posenc := range preferredEncodings {
+		ext := extensionForEncoding(posenc)
+		fname := fpath + ext
+
+		if f.root.Exists("/", fname) {
+			available = append(available, posenc)
+		}
+	}
+
+	// Negotiate the best content encoding to use
+	negenc := negotiateContentEncoding(r, available)
+	if negenc == "" {
+		// If we fail to negotiate anything try the base file
+		return f.openAndStat(fpath)
+	}
+
+	ext := extensionForEncoding(negenc)
+	if file, info, err := f.openAndStat(fpath + ext); err == nil {
+		wHeader := w.Header()
+		wHeader[contentEncodingHeader] = []string{negenc}
+		wHeader.Add(varyHeader, acceptEncodingHeader)
+
+		if len(r.Header[rangeHeader]) == 0 {
+			// If not a range request then we can easily set the content length which the
+			// Go standard library does not do if "Content-Encoding" is set.
+			wHeader[contentLengthHeader] = []string{strconv.FormatInt(info.Size(), 10)}
+		}
+		return file, info, nil
+	}
+
+	// If all else failed, fall back to base file once again
+	return f.openAndStat(fpath)
+}
+
+// Determines the best encoding to use
+func negotiateContentEncoding(r *http.Request, available []string) string {
+	values := strings.Split(r.Header.Get(acceptEncodingHeader), ",")
+	aes := []string{}
+
+	// Clean the values
+	for _, v := range values {
+		aes = append(aes, strings.TrimSpace(v))
+	}
+
+	for _, a := range available {
+		for _, acceptEnc := range aes {
+			if acceptEnc == a {
+				return a
+			}
+		}
+	}
+
+	return ""
+}
+
+// Implements http.Handler
+func (f *gzipFileHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	upath := r.URL.Path
+	if !strings.HasPrefix(upath, "/") {
+		upath = "/" + upath
+		r.URL.Path = upath
+	}
+
+	fpath := path.Clean(upath)
+	if strings.HasSuffix(fpath, "/") {
+		http.NotFound(w, r)
+		return
+	}
+
+	// Find the best acceptable file, including trying uncompressed
+	if file, info, err := f.findBestFile(w, r, fpath); err == nil {
+		http.ServeContent(w, r, fpath, info.ModTime(), file)
+		// Satisy errcheck
```

Comment:
Satisfy*

Reply:
Why not something like `logger.ErrorIfCalling(file.Close)`

Reply:
Fixed